docs/integration-guide.md

```diff
@@ -107,6 +107,239 @@ Here's how we recommend doing it:
+to be consistent with other webhook integrations. We recommend following the
+format `api_webhookname_webhook`.
+
+At minimum, the webhook function must accept `request`, `user_profile`, and
+`client`. You may also want to define additional parameters using the `REQ`
+object.
+
+In the example above, we have defined `payload` which is populated from the
+body of the http request, `stream` with a default of `test` (available by
+default in Zulip dev environment), and `topic` with a default of `Hello World`.
+
+In the body of the function we define the body of the message as `Hello! I am
+happy to be here! :smile: `. The `:smile:` indicates an emoji. Then we append a
+link to the Wikipedia article of the day as provided by the json payload.
+
+Then we send a public (stream) message with `check_send_message`.  Finally, we
+return `json_success()`.
+
+#### Step 2: Create an api endpoint for the webhook
+
+In order for a webhook to be externally available, it must be mapped to a url.
+This is done in `zproject/urls.py`. Look for the lines:
+
+```
+# Incoming webhook URLs
+urls += [
+    # Sorted integration-specific webhook callbacks.
+```
+
+And you'll find the entry for Hello World:
+
+```
+url(r'^api/v1/external/helloworld$',  'zerver.views.webhooks.helloworld.api_helloworld_webhook'),
+```
+
+This tells the Zulip api to call the `api_helloworld_webhook` function in
+`zerver/views/webhooks/helloworld.py` when it receives a request at
+`/api/v1/external/helloworld`.
+
+At this point, if you're following along and/or writing your own Hello World
+webhook, you have written enough code to test your integration.  You can do so
+by using curl on the command line:
+
+```
+curl -X POST -H "Content-Type: application/json" -d '{ "type":"wikipedia", "day":"Wed, 6/1", "featured_title":"Marilyn Monroe", "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe" }' http://localhost:9991/api/v1/external/helloworld\?api_key\=<api_key>
+```
+
+After which you should see:
+```
+{"msg":"","result":"success"}
+```
+
+And a message in Zulip. TODO: add screen shot.
+
+#### Step 3: Create tests and fixtures
+
+Every webhook integraton should have a corresponding test class in
+`zerver/tests/test_hooks.py`.
+
+You should name the class <WebhookName>HookTests and this class should accept
+`WebhookTestCase`. For our HelloWorld webhook, we name the test class
+`HelloWorldHookTests`:
+
+```
+class HelloWorldHookTests(WebhookTestCase):
+    STREAM_NAME = 'test'
+    URL_TEMPLATE = "/api/v1/external/helloworld?&api_key={api_key}"
+    FIXTURE_DIR_NAME = 'hello'
+
+    def test_hello_message(self):
+        expected_subject = u"Hello World";
+        expected_message = u"Hello! I am happy to be here! :smile: \nThe Wikipedia featured article for today is **[Marilyn Monroe](https://en.wikipedia.org/wiki/Marilyn_Monroe)**";
+        self.send_and_test_stream_message('hello', expected_subject, expected_message,content_type="application/x-www-form-urlencoded")
```

Comment:
Should make clear that you should do one test function per message type/fixture supported by the webhook.  Maybe a docstring or comment would be a good place to explain this?